### PR TITLE
Selector select use combo-box as default

### PR DESCRIPTION
**Pull request by gjohansson-ST**
## Proposed change

Use combo-box as default for Selector Select.
Allows to empty select for optional fields.
Originates from discussion in https://github.com/home-assistant/core/pull/84482

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: https://github.com/home-assistant/core/pull/84482
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
This removes other use cases, like disabling certain options. I think we should add an empty option for not required select selectors to be able to clear it.

**Comment by gjohansson-ST** [comment]
> This removes other use cases, like disabling certain options. I think we should add an empty option for not required select selectors to be able to clear it.

I'm not that into frontend so can't argue on that while I would have expected it to work more or less same?
Not too hard to add an empty option to clear the field but from a user experience I think it would simply look nicer to use the same style.

**Comment by epenet** [comment]
@bramkragten, are you suggesting that the empty option should be added by frontend? (with a condition on "optional")
Or are you suggesting that the empty option should be added in core within each integration?

**Comment by eifinger** [comment]
I think another option could be to add a "clear" button to empty the field again. Personally I think it is more intuitive than to select an empty option.
Screenshot 2023-03-30 at 08 55 19

File: src/components/ha-selector/ha-selector-select.ts [hunk 1]
```diff
@@ -194,6 +195,16 @@ export class HaSelectSelector extends LitElement {
           `
         )}
       </ha-select>
+      ${this.value
+        ? html`<ha-svg-icon
```
Comment (bramkragten) [inline src/components/ha-selector/ha-selector-select.ts:199]: You should import `ha-svg-icon`

Comment (bramkragten) [reply src/components/ha-selector/ha-selector-select.ts:199]: And it should be an `ha-icon-button`

Comment (bramkragten) [reply src/components/ha-selector/ha-selector-select.ts:199]: Check the password button: https://github.com/home-assistant/frontend/blob/dev/src/components/ha-selector/ha-selector-text.ts#L76

File: src/components/ha-selector/ha-selector-select.ts [hunk 2]
```diff
@@ -194,6 +195,16 @@ export class HaSelectSelector extends LitElement {
           `
         )}
       </ha-select>
+      ${this.value
```
Comment (bramkragten) [inline src/components/ha-selector/ha-selector-select.ts:198]:
```suggestion
      ${this.value && !this.required
```

**Pull request closed without merging**